pkg/provisioner/reconciler/claim-reconciler/reconiler.go

```diff
@@ -205,13 +210,23 @@ func (r *ObjectBucketClaimReconciler) handleProvisionClaim(key client.ObjectKey,
 	setObjectBucketName(ob, key)
 	ob.Spec.StorageClassName = obc.Spec.StorageClassName
 	ob.Spec.ClaimRef, err = claimRefForKey(key, r.internalClient)
-	ob.SetFinalizers([]string{finalizer})
 	ob.Spec.ReclaimPolicy = options.ReclaimPolicy
+	ob.SetFinalizers([]string{finalizer})
```

Comment:
maybe in another pr the assignments to the ob, creation of the ob, and setting its status can become a helper func just to reduce the length of handleProvision.